webui/react/src/components/DeterminedAuth.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import UIProvider, { DefaultTheme } from 'hew/Theme';
+import { useInitApi } from 'hew/Toast';
+import { useState } from 'react';
+
+import { login } from 'services/api';
+
+import DeterminedAuth, {
+  PASSWORD_ID,
+  SUBMIT_ID,
+  USERNAME_ID,
+  WEAK_PASSWORD_SUBJECT,
+} from './DeterminedAuth';
+import { ThemeProvider } from './ThemeProvider';
+
+const USERNAME = 'username';
+const WEAK_PASSWORD = 'password';
+const STRONG_PASSWORD = 'dO2Ccf5ceozXJFuMhjEx32UgtR4X4yZTm1Vv';
+
+vi.mock('services/api', () => ({
+  login: vi.fn().mockResolvedValue({ token: '', user: {} }),
+}));
+
+const Container = (): JSX.Element => {
+  const [canceler] = useState(new AbortController());
+  useInitApi();
+  return <DeterminedAuth canceler={canceler} />;
+};
+
+const setup = () => {
+  render(
+    <UIProvider theme={DefaultTheme.Light}>
+      <ThemeProvider>
+        <Container />
+      </ThemeProvider>
+    </UIProvider>,
+  );
+};
+
+const user = userEvent.setup();
+describe('DeterminedAuth', () => {
+  it('logs in with strong password', async () => {
+    setup();
+
+    await user.type(screen.getByTestId(USERNAME_ID), USERNAME);
+    await user.type(screen.getByTestId(PASSWORD_ID), STRONG_PASSWORD);
+    user.click(screen.getByTestId(SUBMIT_ID));
+
+    await waitFor(() => {
+      // onFinish begins:
+      expect(screen.getByRole('button')).toBeDisabled();
+    });
+    await waitFor(() => {
+      // onFinish ends:
+      expect(screen.getByRole('button')).not.toBeDisabled();
```

Comment:
nit: this part isn't in the weak password test, it should probably be in both or neither